synfig-studio/src/gui/docks/dock_timetrack2.cpp

```diff
@@ -227,13 +232,42 @@ void Dock_Timetrack2::setup_tool_palette()
 				current_widget_timetrack->set_action_state(action_state);
 		}, *this));
 		action_button_map[tool_button->get_name()] = tool_button;
-		tool_item_group->add(*tool_button);
+		toolbar->append(*tool_button);
+	}
+
+	Gtk::SeparatorToolItem* separator = Gtk::manage(new Gtk::SeparatorToolItem());
+	separator->set_name("separator");
+	toolbar->append(*separator);
+
+	struct InterpolationButtonInfo {
+		std::string name;
+		synfig::Interpolation interpolation;
+	};
+
+	const std::vector<InterpolationButtonInfo> interp_buttons_info{
+		{N_("Clamped"), synfig::INTERPOLATION_CLAMPED},
+		{N_("TCB"), synfig::INTERPOLATION_TCB},
+		{N_("Constant"), synfig::INTERPOLATION_CONSTANT},
+		{N_("Ease In/Out"), synfig::INTERPOLATION_HALT},
+		{N_("Linear"), synfig::INTERPOLATION_LINEAR}
+	};
+
+	for (const auto & interp_button_info: interp_buttons_info) {
+		Gtk::Image* image= Gtk::manage(new Gtk::Image());
+		image->set_from_icon_name(interpolation_icon_name(interp_button_info.interpolation), Gtk::IconSize::from_name("synfig-small_icon_16x16"));
+		Gtk::ToolButton *tool_button = manage(new Gtk::ToolButton(*image, _((interp_button_info.name).c_str())));
+		tool_button->signal_clicked().connect(sigc::track_obj([this, interp_button_info](){
+			current_widget_timetrack->interpolate_selected(interp_button_info.interpolation);
+		}, *this));
+		std::string tool_tip = N_("Change waypoint interpolation to " + interp_button_info.name);
+		tool_button->set_tooltip_text(_(tool_tip.c_str()));

```

Comment:
No, you should split/join string you want to translate.
Different languages have different word order in their grammars.

Use string format instead.

```c++
strprintf(_("Change waypoint interpolation to %s"), interp_button_info.name);
```

Reply:
alright, makes sense. Done, not exactly sure if it is how it should be thougg, so i will leave this open

Reply:
@rodolforg can't I just remove the N_ on line 262 and concactenate the string regularly then use the (gettext) _() on 263 ?

Reply:
No, you can't. It breaks proper translations.

The problem was I typed/wrote the wrong code. This is right:

```
strprintf(_("Change waypoint interpolation to %s"), interp_button_info.name.c_str());
```

Reply:
oh, yes fair enough it now works well.